### PR TITLE
SDK2: Replace old SDK in preview feature controller

### PR DESCRIPTION
**Pull request by bitoku**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-4665

### What this PR does / why we need it:

This PR replaces old SDK in preview feature controller.

### Test plan for issue:

Test preview feature in local dev cluster according to https://learn.microsoft.com/en-us/azure/openshift/howto-enable-nsg-flowlogs.

### Is there any documentation that needs to be updated for this PR?

N/A

### How do you know this will function as expected in production? 

Enable preview feature.

### CODE REVIEW

**Review by kimorris27 — DISMISSED** [review]

**Comment by kimorris27** [comment]
Third E2E run had the same failure as the first - cluster installation failed because k8s API server wasn't ready.

There's no way it's related to your changes, but if it happens again on this fourth try, maybe we should start looking at changes that were merged recently (?).

**Review by kimorris27 — APPROVED** [review]

File: pkg/frontend/adminactions/resources_list_test.go
Comment (kimorris27) [inline pkg/frontend/adminactions/resources_list_test.go:1]: Someone else merged a PR to fix these unit test cases during NASA time (https://github.com/Azure/ARO-RP/pull/3961), so I'm not sure why there's still a diff shown here on your PR... I'd rebase on master once more.

Comment (bitoku) [reply pkg/frontend/adminactions/resources_list_test.go:1]: Thank you! it's gone.

File: pkg/util/azureclient/azuresdk/armnetwork/flowlogs.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	Get(ctx context.Context, resourceGroupName string, networkWatcherName string, flowLogName string, options *armnetwork.FlowLogsClientGetOptions) (result armnetwork.FlowLogsClientGetResponse, err error)
+
+	FlowLogsClientAddons
+}
+
+type flowLogsClient struct {
+	*armnetwork.FlowLogsClient
+}
+
+func NewFlowLogsClient(subscriptionID string, credential azcore.TokenCredential, options *arm.ClientOptions) (FlowLogsClient, error) {
```
Comment (rajdeepc2792) [inline pkg/util/azureclient/azuresdk/armnetwork/flowlogs.go:26]: Reference Comment: https://github.com/Azure/ARO-RP/pull/3619#discussion_r1670895639

We should be returning structs, not interfaces as per the go standard. You can refer the roledefinitions client for the usage https://github.com/Azure/ARO-RP/blob/master/pkg/util/azureclient/azuresdk/armauthorization/roledefinitions.go

For now we can ignore the exisitng clients returning interfaces, but can stop returning the interfaces for new code.

Comment (bitoku) [reply pkg/util/azureclient/azuresdk/armnetwork/flowlogs.go]: Thanks! nice catch!

**Review by jaitaiwan — APPROVED** [review]
LGTM

**Pull request merged**